### ISSUE TITLE
clean larger queue will block the queue

### ISSUE THREAD

**Issue by ericcheww** [OptimalBits/bull#1303]
when the queue's size is 1M and I clean all of them, the queue will block and will throw 'too busy to run' error. It seems like bull queue is not efficient to clear the queue.

**Comment by jamesbillinger** [comment]
So in case it helps anyone, we ran into this as well and made our own js function to clean the queues.  It uses UNLINK rather than DEL and adds a timeout to avoid tying up the server.  I did my best to translate the lua logic back to js.

```
let queues = [
  {
    queue: 'queue1',
    redis: config.redis.redis1
  },
  {
    queue: 'queue2',
    redis: config.redis.redis2
  }
];

let start = new Date();
console.log('starting to clean the queues');
async.each(
  queues,
  (q, callback) => {
    let r = redis.createClient({
      key: config.key,
      ...q.redis
    });
    let h = moment().subtract(1, 'hour');
    async.eachOfSeries(
      ['wait', 'active', 'paused', 'delayed', 'completed', 'failed'],
      (k, ki, callback2) => {
        let verb = 'lrange';
        if (ki > 2) {
          verb = 'zrange';
        }
        r[verb](`bull:${q.queue}:${k}`, 0, -1, (err, items) => {
          if (err) {
            console.log(err);
            callback2();
          } else {
            async.eachSeries(
              items || [],
              (item, callback3) => {
                setTimeout(() => {
                  r.exists(`bull:${q.queue}:${item}:lock`, (err2, e) => {
                    err2 && console.log(err2);
                    if (!e) {
                      r.hget(`bull:${q.queue}:${item}`, 'timestamp', (err3, ts) => {
                        if (moment(parseInt(ts)).isBefore(h)) {
                          if (ki <= 2) {
                            r.lrem(`bull:${q.queue}:${k}`, 0, item);
                          } else {
                            r.zrem(`bull:${q.queue}:${k}`, item);
                          }
                          r.unlink(`bull:${q.queue}:${item}`, (uErr) => {
                            uErr && console.log(uErr);
                            callback3();
                          });
                        } else {
                          callback3();
                        }
                      });
                    } else {
                      callback3();
                    }
                  });
                }, 5);
              },
              callback2
            );
          }
        });
      },
      callback
    );
  },
  () => {
    console.log(`done in ${new Date() - start}ms`);
  }
);
```

**Comment by sedge** [comment]
This is a pretty serious problem, especially for processing jobs in parallel. Trying to clean the queue will cause the other workers to lock up and start throwing errors like mad while Lua finishes its work.

**Comment by manast** [comment]
@sedge you could just pause the queue globally, clean it and then unpause it, that should work I think.

**Comment by manast** [comment]
I am going to close this one since no specific code that reproduces the issue is provided. If relevant please create a new issue.
